backend/hitas/models/apartment.py

```diff
@@ -241,10 +241,13 @@ def sell_by_date(self):
                 if sell_by_date is None:
                     continue
 
-                # If apartment was first sold after it was completed,
-                # the sell by date is calculated based on first sale date.
-                if cos.first_purchase_date is not None and cos.first_purchase_date > sell_by_date:
-                    sell_by_date = cos.first_purchase_date
+                # If apartment was sold after it was completed, the sell by date is calculated based on first sale date.
+                if hasattr(cos, "first_purchase_date"):
```

Comment:
I guess this is because of the [Sentry issue](https://sentry.test.hel.ninja/organizations/city-of-helsinki/issues/2319/?query=is%3Aunresolved&referrer=issue-stream)? In theory, this shouldn't be needed, as the value should always come annotated from `hitas.services.condition_of_sale.condition_of_sale_queryset`. There is even a test for this (`test__api__apartment__list__sell_by_date`), but since it failed once in ninja and I can't replicate the issue, we can add this as a quick fix.

Reply:
Yes exactly. I can replicate it manually locally and in ninja, but I can't seem to replicate the issue with tests. Still something had to be done about it and this fixes it so 🤷